### PR TITLE
Damm ophyd device

### LINKED ISSUE #22: Daamn 2 motor ophyd device @D-hutch

### PR DESCRIPTION
**Pull request by MDecarabas**
- Close #22

### CODE REVIEW

**Comment by prjemian** [comment]
In the original statement (above), I prefixed the `close #22` with a `- `.  Github makes this statement more descriptive by automatically adding the issue's title.

**Comment by prjemian** [comment]
There's a merge conflict in an `__init__.py` file.

File: instrument/devices/damm.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"""
+Damm in station 8-ID-D
```
Comment (prjemian) [inline instrument/devices/damm.py:2]:
```suggestion
Damm (dynamic aperture) in station 8-ID-D
```

I'm guessing this is a dynamic aperture.  Should confirm.  It helps me to translate DAMM from jargon.

File: instrument/devices/damm.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        h_motor: str,
+        v_motor: str,
+        *args,
+        **kwargs,
+    ):
+        # Determine the prefix for the motors
+        pieces = prefix.strip(":").split(":")
+        self.motor_prefix = ":".join(pieces[:-1])
+
+        self._h_motor = h_motor
```
Comment (prjemian) [inline instrument/devices/damm.py:33]: Why name these h & v when these are pitch and yaw motor names?

Comment (MDecarabas) [reply instrument/devices/damm.py:33]: because that is as it was written on the epics screen used by 8-ID

Comment (prjemian) [reply instrument/devices/damm.py:33]: Ooph!

**Review by prjemian — COMMENTED** [review]
I have to hold my nose here.  The naming is not consistent in this class.  The EPICS screen could have its name changed.  If the beamline is OK with it, then I could be, too.  Let's get QZ's or Suresh's attention.

**Comment by MDecarabas** [comment]
Ok, I will ask them when I see them today and relay the information back

**Comment by prjemian** [comment]
@qzhang234 @sureshnaps - Just a question of names here.  There's an inconsistency in the names (.DESC) given to the EPICS motors for pitch & yaw.  Why do the screens show "h" and "v"?  Should we propagate this into the ophyd support?  See my question above.

**Comment by qzhang234** [comment]
@prjemian I'm with Eric at the beamline. Just to confirm, is the question on pitch and yaw related to the DAMM definition?

**Comment by prjemian** [comment]
Yes

**Comment by sureshnaps** [comment]
[Attachment.png] [Attachment_1.png]  [Attachment_2.png]  [Attachment_3.png]  [Attachment_4.png]



damm is just a fixed aperture that can be scanned in X and Y (h and v)



slit with the exception of the 2 in 8ida are all standard 4-blade JJ slits, with added H,V translation underneath to move the whole slit which does not impact the slit database in anyway.



The RL-1 which is CRL-1 is the one that has x,y,pitch, yaw and 10 lens motors - this is separate than the above 2


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by prjemian** [comment]
I did not get any attachments in the GitHub comments.  Sent by email?  Please post them to this PR.

**Pull request merged**